src/components/ColorPicker/components/ColorPickerContent/ColorPickerContentComponent.jsx

```diff
@@ -14,25 +15,64 @@ const ColorPickerContentComponent = ({
   noColorText,
   colorStyle,
   ColorIndicatorIcon,
+  SelectedIndicatorIcon,
   shouldRenderIndicatorWithoutBackground,
-  NoColorIcon
+  NoColorIcon,
+  isBlackListMode,
+  colorsList,
+  isMultiselect
 }) => {
   const onClearButton = useCallback(() => {
     onValueChange(null);
   }, [onValueChange]);
+
+  const onColorClicked = useCallback(
```

Comment:
instead of doing this like this with some heavy logic
please separate it into two hook
1. the callback which just changes the state of the selected colors
2. useEffect which will call the onValueCahnge when the selected colors are changed